enhancements/cluster-logging/logs-based-alerts.md

```diff
@@ -0,0 +1,300 @@
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Logs-based Alerts
+
+## Summary
+
+This document proposes a solution and a delivery plan for pushing logs-based alerts from OpenShift Logging Loki to Cluster-Monitoring AlertManager. It complements the existing [user-workload monitoring stack][uwm-docs] and [alerting stack][alerting-docs], enabling a full self-service experience for platform and workload logs-based monitoring.
+
+## Motivation
+
+Since OpenShift 4.6, application owners can configure alerting rules based on metrics themselves as described in [User Workload Monitoring][user-workload-monitoring-enhancement] (UWM)
+enhancement. The rules are defined as `PrometheusRule` resources and can be based on platform and/or application metrics.
+
+To expand the alerting capabilities on logs as an observability signal, cluster admins and application owners should be able to configure alerting rules as described in the [Loki Rules][loki-rules-docs] docs and in the [Loki Operator Ruler][loki-operator-ruler-support] upstream enhancement.
+
+[AlertingRule][alertingrule-crd] CRD fullfills the requirement to define alerting rules for Loki similar to `PrometheusRule`.
+
+[RulerConfig][rulerconfig-crd] CRD fullfills the requirement to connect the Loki Ruler component to notify a list of Prometheus AlertManager hosts on firing alerts.
+
+### User Stories
+
+* As an application owner, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from my workload logs.
+* As a cluster admin, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from the platform logs.
+* As a cluster admin, I want to use AlertingRule custom resources to notify AlertManager on firing alerts from the audit logs.
+
+### Goals
+
+* Application Owners can configure alert notifications for application alerts based on workload logs.
+* Cluster admins can configure alert notifications for platform alerts based on platform logs.
+* Cluster admins can configure alert notifications for platform alerts based on audit logs.
+
+### Non-Goals
+
+* Specific alerting support on the OCP console for Logs-based alerts. This should be seamless in the existing alerting UIs.
+* Recording rules to compile metrics from logs.
+
+## Proposal
+
+We plan to leverage the `AlertingRule` and `RulerConfig` custom resource definitions already exposed by the Loki Operator so that:
+- cluster admins can configure how the platform logs can notify on alertworthy platform and/or audit logs.
+- application owners can configure how the workload logs can notify on alertworthy logs.
+
+### Workflow Description
+
+#### Logs-based Alerts for Application Owners
+
+1. The application owner creates an AlertingRule custom resource on a workload project, e.g.
+
+```yaml
+apiVersion: loki.grafana.com/v1beta1
+kind: AlertingRule
+metadata:
+  name: my-workload-alerts
+  namespace: my-workload-ns
+  labels:
+    openshift.io/cluster-monitoring: "true"
```

Comment:
I expect that we don't need this label?

Reply:
The Loki-Operators follows the same pattern as the Prometheus-Operator for detecting AlertingRules/RecordingRules based on selectors and namespaceSelectors. I would say keeping the same labels to detect Loki AlertingRules/RecordingRules in OpenShift Logging as you do with PrometheusRules in OpenShift Monitoring, would make it easier for users for both observability pieces. WDYT?